metadata-ingestion/src/datahub/emitter/mcp.py

```diff
@@ -104,16 +130,75 @@ def validate(self) -> bool:
             return False
         return True
 
-    def to_obj(self, tuples: bool = False) -> dict:
-        return self.make_mcp().to_obj(tuples=tuples)
+    def to_obj(self, tuples: bool = False, for_file: bool = False) -> dict:
+        obj = self.make_mcp().to_obj(tuples=tuples)
+        if for_file:
+            # Undo the double JSON serialization that happens in the MCP aspect.
+            if (
+                obj.get("aspect")
+                and obj["aspect"].get("contentType") == "application/json"
+            ):
+                obj["aspect"] = {"json": json.loads(obj["aspect"]["value"])}
```

Comment:
What are the implications of doing this? Are any backwards compatibilities broken?

Reply:
no this is backwards compatible, since the source has been updated to read either format

I added a flag to the file sink to preserve the old behavior